pkg/exec/helpers.go

```diff
@@ -110,10 +110,11 @@ func RunWithStdoutReader(cmd Cmd, readerFunc func(io.Reader) error) error {
 	}()
 
 	err = cmd.Run()
+	pw.Close()
+	err2 := <-errChan
```

Comment:
why moving it here? I guess it can block the func if `err!=nil`, can't it?

Reply:
Yes, we should be waiting. We didn't wait before because it could hang. That _should_ be noticed and caught because it means a bug. In this case the pipe still being open.

This was the original impl (wait for both) and then we had a quick fix the prevent a hang by moving it 😛

Compare to os/exec.